diff --git a/node_modules/@stylexjs/webpack-plugin/src/index.js b/node_modules/@stylexjs/webpack-plugin/src/index.js
index 5b2c270..40557ac 100644
--- a/node_modules/@stylexjs/webpack-plugin/src/index.js
+++ b/node_modules/@stylexjs/webpack-plugin/src/index.js
@@ -72,9 +72,9 @@ class StylexPlugin {
         (loaderContext, module) => {
           if (
             // JavaScript (and Flow) modules
-            /\.jsx?/.test(path.extname(module.resource)) ||
+            /^\.jsx?$/.test(path.extname(module.resource)) ||
             // Typescript modules
-            /\.tsx?/.test(path.extname(module.resource))
+            /^\.tsx?$/.test(path.extname(module.resource))
           ) {
             // It might make sense to use .push() here instead of .unshift()
             // Webpack usually runs loaders in reverse order and we want to ideally run
